geoips/interfaces/module_based/readers.py

```diff
@@ -19,5 +24,233 @@ class ReadersInterface(BaseModuleInterface):
+        metadata_only : bool, default=False
+            * Return before actually reading data if True
+        chans : list of str, default=None
+            * List of desired channels (skip unneeded variables as needed).
+            * Include all channels if None.
+        area_def : pyresample.AreaDefinition, default=None
+            * Specify region to read
+            * Read all data if None.
+        self_register : str or bool, default=False
+            * register all data to the specified dataset id (as specified in the
+              return dictionary keys).
+            * Read multiple resolutions of data if False.
+
+        Returns
+        -------
+        dict of xarray.Datasets
+            * dictionary of xarray.Dataset objects with required Variables and
+              Attributes.
+            * Dictionary keys can be any descriptive dataset ids.
+
+        See Also
+        --------
+        :ref:`xarray_standards`
+            Additional information regarding required attributes and variables
+            for GeoIPS-formatted xarray Datasets.
+        """
+        start_times = [
+            dts[0] for dts in all_metadata["METADATA"].attrs["source_file_datetimes"]
+        ]
+        end_times = [
+            dts[1] for dts in all_metadata["METADATA"].attrs["source_file_datetimes"]
+        ]
+        stimes = list(set(start_times))
+        etimes = list(set(end_times))
+        import collections
+
+        ingested_xarrays = collections.defaultdict(list)
+        for time in stimes:
+            scan_time_files = [dt == time for dt in start_times]
+            # Call the associated reader for a single file
+            data_dict = call_single_file_func(
+                np.array(fnames)[scan_time_files],
+                metadata_only=metadata_only,
+                chans=chans,
+                area_def=area_def,
+                self_register=self_register,
+            )
+            for (
+                dname,
+                dset,
+            ) in data_dict.items():
+                ingested_xarrays[dname].append(dset)
+
+        if len(stimes) == 1:
+            # No need to stack if we are only reading in one scan time
+            # This is likely temporary to maintain backwards compatibility
+
+            # This is not hit if we are provided multiple scan times.
+            return data_dict
+
+        import xarray
+
+        # merged_dset = xarray.Dataset()
+        # Now that we've ingested all scan times, stack along time dimension
+        metadata = all_metadata["METADATA"]
+        dict_xarrays = {}
+        for dname, list_xarrays in ingested_xarrays.items():
+            if dname == "METADATA":
+                continue
+            merged_dset = xarray.concat(list_xarrays, dim="time")
+            merged_dset.attrs["start_datetime"] = min(stimes)
+            merged_dset.attrs["end_datetime"] = max(etimes)
+            merged_dset = merged_dset.assign_coords({"time": stimes})
```

Comment:
Is this suggesting that our xarray standard should be to use a "time" dimension/coord for the 3rd dataset dimension?  Does it also imply that the 3rd variable dimension will always be time?  How do we add in vertical level in the future for 4d datasets?

Also, should we update our x/y coords/dimensions to actually be named latitude and longitude to match the variable names ?  I expect this is fine for now as is, but we need to finalize/formalize the xarray standards, and then revisit to ensure we are consistent.

Reply:
FWIW cf-conventions for model data suggest that datasets are layered in the following order: "date or time", "height or depth", "latitude", or "longitude"

[NetCDF Climate and Forecast (CF) Metadata Conventions](https://cfconventions.org/cf-conventions/cf-conventions.html#dimensions)

Reply:
I say we go with the cf-conventions and reflect that in our code / standards. We'll have to update this in the case we run into 4d datasets. We can either create an issue, or revisit this point once necessary.

Reply:
Does this work with sectoring at the moment?

Reply:
Could you please create an issue for finalizing xarray/reader standards (if one doesn't already exist), link to this conversation from the issue, link to the issue from this conversation, then you can resolve this conversation?

Reply:
See #817. Also, see my [comment below about sectoring](https://github.com/NRLMMD-GEOIPS/geoips/pull/427#issuecomment-2457730660) @AJThorpe.